xbmc/platform/darwin/ios-common/AnnounceReceiver.mm

```diff
@@ -116,70 +120,80 @@ void AnnounceBridge(ANNOUNCEMENT::AnnouncementFlag flag, const char *sender, con
     {
       bool needsRecaching;
       std::string cachedThumb(CTextureCache::GetInstance().CheckCachedImage(thumb, needsRecaching));
-      //LOG("thumb: %s, %s", thumb.c_str(), cachedThumb.c_str());
       if (!cachedThumb.empty())
       {
         std::string thumbRealPath = CSpecialProtocol::TranslatePath(cachedThumb);
-        [item setValue:@(thumbRealPath.c_str()) forKey:@"thumb"];
+        item[@"thumb"] = @(thumbRealPath.c_str());
       }
     }
     double duration = g_application.GetTotalTime();
     if (duration > 0)
-      [item setValue:@(duration) forKey:@"duration"];
-    [item setValue:@(g_application.GetTime()) forKey:@"elapsed"];
+      item[@"duration"] = @(duration);
+    item[@"elapsed"] = @(g_application.GetTime());
     int current = CServiceBroker::GetPlaylistPlayer().GetCurrentSong();
     if (current >= 0)
     {
-      [item setValue:@(current) forKey:@"current"];
-      [item setValue:@(CServiceBroker::GetPlaylistPlayer().GetPlaylist(CServiceBroker::GetPlaylistPlayer().GetCurrentPlaylist()).size()) forKey:@"total"];
+      item[@"current"] = @(current);
+      item[@"total"] = @(CServiceBroker::GetPlaylistPlayer()
+                             .GetPlaylist(CServiceBroker::GetPlaylistPlayer().GetCurrentPlaylist())
+                             .size());
     }
     if (g_application.CurrentFileItem().HasMusicInfoTag())
     {
-      const std::vector<std::string> &genre = g_application.CurrentFileItem().GetMusicInfoTag()->GetGenre();
+      const auto genre = g_application.CurrentFileItem().GetMusicInfoTag()->GetGenre();
       if (!genre.empty())
       {
-        NSMutableArray *genreArray = [[NSMutableArray alloc] initWithCapacity:genre.size()];
-        for(std::vector<std::string>::const_iterator it = genre.begin(); it != genre.end(); ++it)
+        auto genreArray = [[NSMutableArray alloc] initWithCapacity:genre.size()];
+        for (auto it = genre.cbegin(); it != genre.cend(); ++it)
         {
           [genreArray addObject:@(it->c_str())];
         }
-        [item setValue:genreArray forKey:@"genre"];
+        item[@"genre"] = genreArray;
       }
     }
-    //LOG(@"item: %@", item.description);
-    [g_xbmcController performSelectorOnMainThread:@selector(onPlay:) withObject:item  waitUntilDone:NO];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [g_xbmcController.MPNPInfoManager onPlay:item];
+    });
   }
   else if (msg == "OnSpeedChanged" || msg == "OnPause")
   {
-    NSDictionary *item = [dict valueForKey:@"item"];
-    NSDictionary *player = [dict valueForKey:@"player"];
-    [item setValue:[player valueForKey:@"speed"] forKey:@"speed"];
-    [item setValue:@(g_application.GetTime()) forKey:@"elapsed"];
-    //LOG(@"item: %@", item.description);
-    [g_xbmcController performSelectorOnMainThread:@selector(OnSpeedChanged:) withObject:item  waitUntilDone:NO];
+    id item = dict[@"item"];
+    id player = dict[@"player"];
+    item[@"speed"] = player[@"speed"];
```

Comment:
doesn't this line spit a warning? I think it's better to explicitly type `item` and `player` as `NSDictionary*`. Also seems like a copy-paste from above.